gameobjects.py

```diff
@@ -9,12 +10,31 @@ def __init__(self, wagon_data, position, image):
         self.wagon_data = wagon_data
         self.target = None
         self.position = position
-        self.rect.center = position
+        self.rect.topleft = position
         self.speed = 5
+        self.track = None
 
     def set_target(self, target):
         self.target = target
 
+    def on_begin_drag(self, pos):
+        super().on_begin_drag(pos)
+
+    def on_end_drag(self):
+        super().on_end_drag()
+        new_track = self.colliding_track()
+        if new_track is None:
+            self.rect.x = self.ghost_sprite.rect.x
+            self.rect.y = self.ghost_sprite.rect.y
+        else:
+            if new_track is not self.track and new_track.full() is False:
```

Comment:
"is not self.track" does that mean you can not sort within the track you are currently on? I like it, was just wondering if that is the case